vagrant/provisioning/roles/confluent-platform-install/templates/ksql-server.properties

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright 2018 Confluent Inc.
+#
+# Licensed under the Confluent Community License (the "License"); you may not use
+# this file except in compliance with the License.  You may obtain a copy of the
+# License at
+#
+# http://www.confluent.io/confluent-community-license
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations under the License.
+#
+
+#------ Endpoint config -------
+
+### HTTP  ###
+# The URL the KSQL server will listen on:
+# The default is any IPv4 interface on the machine.
+# NOTE: If set to wildcard or loopback set 'advertised.listener' to enable pull queries across machines
+listeners=http://0.0.0.0:8088
+
+# Use the 'listeners' line below for any IPv6 interface on the machine.
+# listeners=http://[::]:8088
+
+# If running a multi-node cluster across multiple machines and 'listeners' is set to a wildcard or loopback address
+# 'advertised.listener' must be set to the URL other KSQL nodes should use to reach this node.
+# advertised.listener=?
+
+### HTTPS ###
+# To switch KSQL over to communicating using HTTPS comment out the 'listeners' line above
+# uncomment and complete the properties below.
+# See: https://docs.confluent.io/current/ksql/docs/installation/server-config/security.html#configuring-ksql-cli-for-https
+#
+# listeners=https://0.0.0.0:8088
+# advertised.listener=?
+# ssl.keystore.location=?
+# ssl.keystore.password=?
+# ssl.key.password=?
+
+#------ Logging config -------
+
+# Automatically create the processing log topic if it does not already exist:
+ksql.logging.processing.topic.auto.create=true
+
+# Automatically create a stream within KSQL for the processing log:
+ksql.logging.processing.stream.auto.create=true
+
+# Uncomment the following if you wish the errors written to the processing log to include the
+# contents of the row that caused the error.
+# Note: care should be taken to restrict access to the processing topic if the data KSQL is
+# processing contains sensitive information.
+#ksql.logging.processing.rows.include=true
+
+#------ External service config -------
+
+# The set of Kafka brokers to bootstrap Kafka cluster information from:
+bootstrap.servers=localhost:{{ kafka_port | default(9092) }}
+
+# uncomment the below to start an embedded Connect worker
+# ksql.connect.worker.config=config/connect.properties
+# ksql.connect.configs.topic=ksql-connect-configs
+
+# Uncomment and complete the following to enable KSQL's integration to the Confluent Schema Registry:
```

Comment:
Sometimes it's hard to decide whether to template, or to replace/lineinfile/blockinfile the default configuration.  With templates, on each new confluent-platform release, we have to refresh the template ... with lineinfile sometimes we have to adjust the replacements, but usually existing property keys don't change too often.

Reply:
There is a lot of work to make this configurable. All these can run on separate servers, so we'll need to introduce variables. Then some of the services need to have their data and log folders setup. But, currently I don't know which ones. I've setup several that I had to, because of errors.